hcl/hcl.go

```diff
@@ -719,34 +727,51 @@ func parseGenerateHCLBlock(block *ast.Block) (GenHCLBlock, error) {
 	var (
 		lets    hclsyntax.Blocks
 		content *hclsyntax.Block
+		asserts []AssertConfig
 	)
 
 	err := validateGenerateHCLBlock(block)
 	if err != nil {
 		return GenHCLBlock{}, err
 	}
 
-	for _, b := range block.Body.Blocks {
-		switch b.Type {
+	errs := errors.L()
+
+	for _, subBlock := range block.Blocks {
+		switch subBlock.Type {
 		case "lets":
-			lets = append(lets, b)
+			lets = append(lets, subBlock.Block)
+		case "assert":
+			// TODO(KATCIPIS): error handling tests
+			assertCfg, err := parseAssertConfig(subBlock)
+			if err != nil {
+				errs.Append(err)
+				continue
+			}
+			asserts = append(asserts, assertCfg)
 		case "content":
 			if content != nil {
-				return GenHCLBlock{}, errors.E(b.Range(),
+				errs.Append(errors.E(subBlock.Range(),
 					"multiple generate_hcl.content blocks defined",
-				)
+				))
+				continue
 			}
-			content = b
+			content = subBlock.Block
 		default:
 			// already validated but sanity checks...
-			panic("unreachable")
+			panic(errors.E("terramate internal error: unexpected block type %s", subBlock.Type))
```

Comment:
nice, maybe we can standardize this for internal panics by creating an `errors.Kind` of "terramate internal error".